test/functional/p2p_timeouts.py

```diff
@@ -48,10 +48,13 @@ def run_test(self):
         self.mock_time = int(time.time())
         self.mock_forward(0)
 
-        # Setup the p2p connections
-        no_verack_node = self.nodes[0].add_p2p_connection(TestP2PConn(), wait_for_verack=False)
-        no_version_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
-        no_send_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
+        # Setup the p2p connections, making sure the connections are established before the mocktime is bumped
+        with self.nodes[0].assert_debug_log(['Added connection peer=0']):
+            no_verack_node = self.nodes[0].add_p2p_connection(TestP2PConn(), wait_for_verack=False)
+        with self.nodes[0].assert_debug_log(['Added connection peer=1']):
+            no_version_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
+        with self.nodes[0].assert_debug_log(['Added connection peer=2']):
```

Comment:
would an alternative be to connect no_verack_node last?

Reply:
Yes, I actually thought about that alternative, but relying on the order seemed a bit ugly to me too (and it would require changing  later `assert_debug_log` because the peer ids would change). But I don't have a strong preference, would you prefer that solution?

Reply:
Or yet another alternative would be to append a full dummy connection for the sole purpose of syncing (wouldn't change the ids).

Personally I think any solution is fine.

Reply:
ok, then I'll keep it as is for now.